### PR TITLE
Issue #12351: Fix action 'Bump version/Update Milestone' failing to update milestones

### LINKED ISSUE #12351: Action 'Bump version/Update Milestone' is failing to update milestones
https://github.com/checkstyle/checkstyle/actions/runs/3356052486/jobs/5560809102

```
Run .ci/update-github-milestone.sh 10.4
Error: Define GITHUB_TOKEN environment variable
Error: Process completed with exit code 1.
```

but commit is pushed - https://github.com/checkstyle/checkstyle/commit/9b4e1a6eec674ca1108f6aaa64f3616532718c98 ,  but milestone is failed to be updated

commit `minor: Bump release version to 10.4` was very confusing to me, we should rename it be `minor: Bump release version to 10.4-SNAPSHOT`

### PR DESCRIPTION
**Pull request by stoyanK7**
Resolves #12351

### CODE REVIEW

**Comment by stoyanK7** [comment]
This is just draft. Will remove github actions commit when ready to merge.\

https://github.com/stoyanK7/checkstyle/actions/runs/3440968484

**Comment by stoyanK7** [comment]
It seems that 3d65e6cd62 created these two jobs `bump version/update milestone` and env variable name was initially `READ_ONLY_TOKEN` in `.ci/update-github-milestone.sh`. And then 8bb6f1600df changed all `READ_ONLY_TOKEN` to `GITHUB_TOKEN` in that file but forgot to change it also in `.github/workflows/bump-version-and-update-milestone.yml`.

File: .github/workflows/bump-version-and-update-milestone.yml [hunk 1]
```diff
@@ -38,4 +38,4 @@ jobs:
         run: |
           .ci/update-github-milestone.sh ${{ github.event.inputs.version }}
         env:
-          READ_ONLY_TOKEN: ${{ github.token }}
+          GITHUB_TOKEN: ${{ github.token }}
```
Comment (romani) [inline .github/workflows/bump-version-and-update-milestone.yml:41]: Please do me favor and move env block above run.

**Review by romani — CHANGES_REQUESTED** [review]
Please remove extra commit.
Please change message as I suggested in issue description.

Be ready for merge

File: .github/workflows/bump-version-and-update-milestone.yml (hunk 1 above)
Comment (stoyanK7) [reply .github/workflows/bump-version-and-update-milestone.yml:40]: Done. Changed the commit message as suggested in issue description as well.

File: .github/workflows/bump-version-and-update-milestone.yml [hunk 2]
```diff
@@ -32,10 +32,11 @@ jobs:
         run: |
           git config user.name 'github-actions[bot]'
           git config user.email 'github-actions[bot]@users.noreply.github.com'
-          git commit -am "minor: Bump release version to ${{ github.event.inputs.version }}"
+          git commit -am\ 
+            "minor: Bump release version to ${{ github.event.inputs.version }}-SNAPSHOT"
```
Comment (romani) [inline .github/workflows/bump-version-and-update-milestone.yml:36]: Please remove 'release' word

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .github/workflows/bump-version-and-update-milestone.yml (hunk 2 above)
Comment (stoyanK7) [reply .github/workflows/bump-version-and-update-milestone.yml]: Done.

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**